examples/src/standard-nft/my-nft.ts

```diff
@@ -0,0 +1,219 @@
+import { NonFungibleToken } from "near-contract-standards/lib";
```

Comment:
Looks like a lot of work for somebody to use a standard token. I wander if there is a way not to override most of the functions if they remain "standard". (just a discussion for the future)

Reply:
Yeah, this need more discussion. In rust it's done via a few macros: `impl_non_fungible_token_core` etc. I didn't see actively maintainted or commonly used JS macro / code generator libraries to integrate with our build pipeline. One way maybe making a "macro" with babel, like our `includeBytes`

Reply:
i completely agree with @volovyks but i think that leads to deeper discussions about the codegen / transpilation process and _why_ something simpler may not work. I think that the desirable outcome is something like this

in this repo (or some other repo) the standard exists and is implemented

```typescript
@NearBindgen({ requireInit: true })
export class NonFungibleToken {
...
}
```

later, someone wants to customize (such as myself) and add a function, called, say, deposit to extend the abilities of the NFT as such 

```typescript
@NearBindgen({ requireInit: true })
export class MyNonFungibleToken extends NonFungibleToken {
    // i get everything from NonFungibleToken by default but now i can extend
   @call({ payable: true})
    deposit(args: DepositArgs): void { ... }
}
```

does that make sense?

we tried that but currently contracts can't be composed in this way, I don't have the errors at hand but I think a lot of it goes to the general codegen process related to the transpilation.

I can write a ticket to discuss further or we can chat in discord etc

Reply:
Yes, I have something like this in mind as well. @waynenilsen issues with described problematics would be great!

Reply:
I merged this one and created #290 . @waynenilsen feel free to experiment and may be create some proof of concept from latest `develop`!